web/forms/implementing_simple_autocompletion/match_element.dart

```diff
@@ -0,0 +1,50 @@
+@HtmlImport('match_element.html')
+library web.forms.implementing_simple_autocompletion_match;
+
+import 'dart:html';
+
+import 'package:polymer/polymer.dart';
+
+@CustomTag('match-element')
+class MatchElement extends PolymerElement {
+
+  /// Candidate string.
+  ///
+  /// The matching string which we want to display highlighted.
+  @published String value;
+
+  /// Input query.
+  ///
+  /// The query whose first match in `value` we want to display highlighted.
+  @published String inputQuery;
+
+  NodeValidator nodeValidator;
+
+  MatchElement.created() : super.created() {
+    nodeValidator = new NodeValidatorBuilder()
+      ..allowTextElements()
+      ..allowElement('strong');
+  }
+
+  void valueChanged(String old) {
+    injectBoundHtml(highlightedValue, element: $['container'], validator: nodeValidator);
```

Comment:
You could store the result of highlightedValue in a property and just use a binding in the template, but I am not opposed to doing it this way.